lib/plugins/blocks.js

```diff
@@ -275,18 +301,28 @@ function inject (bot, { version, storageBuilder }) {
   bot._client.on('update_light', (packet) => {
     let column = bot.world.getColumn(packet.chunkX, packet.chunkZ)
     if (!column) {
-      column = new Chunk()
+      column = createNewChunk()
       bot.world.setColumn(packet.chunkX, packet.chunkZ, column)
     }
 
-    column.loadLight(packet.data, packet.skyLightMask, packet.blockLightMask, packet.emptySkyLightMask, packet.emptyBlockLightMask)
+    // Before 1.13, light packet contained combined skylight and blocklight data
+    // Now they are stored separately in their exclusive fields, and their type are arrays
+    if (packet.data) {
+      column.loadLight(packet.data, packet.skyLightMask, packet.blockLightMask, packet.emptySkyLightMask, packet.emptyBlockLightMask)
+    } else {
+      // Use new method for handling 1.13+ lighting data
+      const flatSkyLightMask = flattenMaskArray(packet.skyLightMask)
```

Comment:
Flatten should not be needed. If it is needed, then the format of chunk packet is wrong.

Reply:
I don't really know what you mean by "wrong", because format of the packet is correct and matches what the game actually sends (and for that particular field, game sends varint length prefixed array of 64-bit integers)

Reply:
Same applies to section mask in map_chunk

Reply:
yes my point is that if we need to unparse what protodef just parsed at every time we use it, then the format we used is useless, and we should for example return a buffer and parse it directly in pchunk
mineflayer should not do any chunk parsing

Reply:
solved